actors/market/tests/market_actor_test.rs

```diff
@@ -1498,6 +1498,186 @@ fn fail_when_deal_update_epoch_is_in_the_future() {
     // TODO: actor.checkState
 }
 
+#[test]
+fn crontick_for_a_deal_at_its_start_epoch_results_in_zero_payment_and_no_slashing() {
+    let start_epoch = ChainEpoch::from(50);
+    let end_epoch = start_epoch + 200 * EPOCHS_IN_DAY;
+    let sector_expiry = end_epoch + 100;
+
+    let client_addr = Address::new_id(CLIENT_ID);
+    let provider_addr = Address::new_id(PROVIDER_ID);
+    let owner_addr = Address::new_id(OWNER_ID);
+    let worker_addr = Address::new_id(WORKER_ID);
+    let control_addr = Address::new_id(CONTROL_ID);
+
+    // set start epoch to coincide with processing (0 + 0 % 2880 = 0)
+    let start_epoch = 0;
+    let mut rt = setup();
+    let deal_id = publish_and_activate_deal(
+        &mut rt,
+        client_addr,
+        provider_addr,
+        owner_addr,
+        worker_addr,
+        control_addr,
+        start_epoch,
+        end_epoch,
+        0,
+        sector_expiry,
+    );
+
+    // move the current epoch to processing epoch
+    let current = process_epoch(start_epoch, deal_id);
+    rt.set_epoch(current);
+    let (pay, slashed) =
+        cron_tick_and_assert_balances(&mut rt, client_addr, provider_addr, current, deal_id);
+    assert_eq!(TokenAmount::from(0u8), pay);
+    assert_eq!(TokenAmount::from(0u8), slashed);
+
+    // deal proposal and state should NOT be deleted
+    get_deal_proposal(&mut rt, deal_id);
+    get_deal_state(&mut rt, deal_id);
+    // TODO: actor.checkState(rt)
+}
+
+#[test]
+fn slash_a_deal_and_make_payment_for_another_deal_in_the_same_epoch() {
+    let start_epoch = ChainEpoch::from(50);
+    let end_epoch = start_epoch + 200 * EPOCHS_IN_DAY;
+    let sector_expiry = end_epoch + 100;
+
+    let client_addr = Address::new_id(CLIENT_ID);
+    let provider_addr = Address::new_id(PROVIDER_ID);
+    let owner_addr = Address::new_id(OWNER_ID);
+    let worker_addr = Address::new_id(WORKER_ID);
+    let control_addr = Address::new_id(CONTROL_ID);
+
+    let mut rt = setup();
+
+    let deal_id1 = publish_and_activate_deal(
+        &mut rt,
+        client_addr,
+        provider_addr,
+        owner_addr,
+        worker_addr,
+        control_addr,
+        start_epoch,
+        end_epoch,
+        0,
+        sector_expiry,
+    );
+    let d1 = get_deal_proposal(&mut rt, deal_id1);
+
+    let deal_id2 = publish_and_activate_deal(
+        &mut rt,
+        client_addr,
+        provider_addr,
+        owner_addr,
+        worker_addr,
+        control_addr,
+        start_epoch + 1,
+        end_epoch + 1,
+        0,
+        sector_expiry,
+    );
+
+    // slash deal1
+    let slash_epoch = process_epoch(start_epoch, deal_id2) + ChainEpoch::from(100);
+    rt.set_epoch(slash_epoch);
+    terminate_deals(&mut rt, provider_addr, &[deal_id1]);
+
+    // cron tick will slash deal1 and make payment for deal2
+    let current = slash_epoch + ChainEpoch::from(1);
```

Comment:
This epoch incrementing from the go test is a noop that obfuscates what's going on.  Keep the runtime at the slash_epoch instead of incrementing here.